noether_tpp/src/tool_path_modifiers/blending_modifiers.cpp

```diff
@@ -0,0 +1,89 @@
+#include <noether_tpp/tool_path_modifiers/blending_modifiers.h>
+
+namespace noether
+{
+AngledOrientationModifier::AngledOrientationModifier(double angle_offset, double tool_radius, bool flip_sign)
+  : angle_offset_(angle_offset), tool_radius_(tool_radius), flip_sign_(flip_sign)
+{
+}
+
+ToolPaths AngledOrientationModifier::modify(ToolPaths tool_paths) const
+{
+  double sign = 1.0;
+
+  for (ToolPath& tool_path : tool_paths)
+  {
+    for (ToolPathSegment& segment : tool_path)
+    {
+      for (Eigen::Isometry3d& waypoint : segment)
+      {
+        waypoint.rotate(Eigen::AngleAxisd(sign * -angle_offset_, Eigen::Vector3d::UnitY())).translate(Eigen::Vector3d(sign * tool_radius_,0,0));
+      }
+    }
+
+    if(flip_sign_)
+      sign *= -1.0;
+  }
```

Comment:
In retrospect, I think the `flip_sign_` parameter is confusing. I think we should design this in such a way that the angle offset is applied correctly with respect to the nominal direction of travel

```suggestion
  for (ToolPath& tool_path : tool_paths)
  {
    Eigen::Vector3d dir = estimateToolPathDirection(tool_path);
    const Eigen::Isometry3d& first = tool_path.at(0).at(0);
    const Eigen::Vector3d& y = first.matrix().col(1).head<3>();
    const Eigen::Vector3d& z = first.matrix().col(2).head<3>();
    // Sign is determined by whether the first point's y-axis is aligned with the nominal path y-axis (i.e., the cross of the waypoint's z-axis with the nominal path direction)
    double sign = z.cross(dir).dot(y) > 0.0 ? 1.0 : -1.0;
    for (ToolPathSegment& segment : tool_path)
    {
      for (Eigen::Isometry3d& waypoint : segment)
      {
        waypoint.rotate(Eigen::AngleAxisd(sign * -angle_offset_, Eigen::Vector3d::UnitY())).translate(Eigen::Vector3d(sign * tool_radius_,0,0));
      }
    }
  }
```

`estimateToolPathDirection` exists [here](https://github.com/ros-industrial/noether/blob/c7c1d0a170318b2970fc45869a3e1eb3d5786a54/noether_tpp/src/tool_path_modifiers/organization_modifiers.cpp#L8)